src/extensionApi.ts

```diff
@@ -407,31 +413,35 @@ export class CommandHandler {
         }
     }
 
-    private async checkExtension(debugInfo: Protocol.CommandLineDetails): Promise<string> {
-        if (debugInfo && debugInfo.properties['debug.details.type'].indexOf('java') >= 0) {
-            if (vscode.extensions.getExtension('vscjava.vscode-java-debug') === undefined) {
-                return 'Debugger for Java extension is required. Install/Enable it before proceeding.';
-            }
-        } else {
-            return `Vscode-Adapters doesn\'t support debugging with ${debugInfo.properties['debug.details.type']} language at this time.`;
+    private async checkExtension(debugInfo: DebugInfo): Promise<string> {
+        if (!debugInfo.isJavaType()) {
```

Comment:
we should also check if debugInfo is undefined, otherwise it's going to fail and show a different error "debugInfo doesn't have isjavaType poperty"